UnitySDK/Assets/ML-Agents/Scripts/RewardProvider/TypedRewardProviderComponent.cs

```diff
@@ -0,0 +1,23 @@
+namespace MLAgents.RewardProvider
+{
+    /// <summary>
+    /// A typed reward provider that provides easy, typed access to RewardProvider implementations.
+    /// Subclasses should
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public class TypedRewardProviderComponent<T> : BaseRewardProviderComponent
+    where T : IRewardProvider, new()
+    {
+        T m_TypedRewardProvider = new T();
```

Comment:
This assumes the constructor does not take arguments

Reply:
yes

Reply:
I don't really see a case for a reward provider needing anything passed into its constructor.  It seems like the low level reward provider will be the backing for most, if not all, RewardProviderComponents.